src/components/dialogs/PreferencesDialog/index.js

```diff
@@ -78,4 +78,13 @@ class PreferencesDialog extends Component {
   }
 }
 
+function parseAsNumberIfNumber (str) {
```

Comment:
This is also a canditate for extraction. Consider using lodash `isInteger` and `isNumber` methods:

https://lodash.com/docs/4.17.4#isInteger

Also conversion methods:

https://lodash.com/docs/4.17.4#toInteger

Beware of returning `undefined`.